tower-http/src/services/fs/serve_dir.rs

```diff
@@ -124,8 +173,27 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
                     HeaderValue::from_str(mime::APPLICATION_OCTET_STREAM.as_ref()).unwrap()
                 });
 
+            if let Some(file_extension) =
+                negotiated_encoding.and_then(|encoding| encoding.to_file_extention())
```

Comment:
Just checking, but `to_file_extension` only returns `None` for "identity" encoding meaning technically `.filter(|encoding| *encoding != Encoding::Identity);` also handles the filtering. I don't necessarily think we have to change it, just making sure I understand why we're doing this check.

Reply:
I actually didn't have the filter before but the problem was the negotiated_encoding will then be passed as `Some(Encoding::Identity)` to the ResponseFuture which means the content encoding header would be set to `identity`. That should be fine but thought it would be nice to not change the behavior when there is no compression.

Reply:
Ah yes that makes sense!